### ISSUE TITLE
dagger.connect takes too long

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#4456]
## Problem

Opening a Dagger session from a client library takes too long.

This has been reported twice in the last week.

## Solution

Make it faster :)

**Comment by kokizzu** [comment]
or at least show the status/progress of what it's currently doing (downloading docker image)

**Comment by helderco** [comment]
> or at least show the status/progress of what it's currently doing (downloading docker image)

I experimented with that in [Nov 17 community call](https://dagger-io.zoom.us/rec/play/L24RATHeqJ6juqlUYJchr65Bqg34zI_paGUfcOicvMlkja6eWmoT5qztjueODWjcHtuhp2fOgeipuxJD.iwLvJRyHrHpZQSxZ?continueMode=true&_x_zm_rtaid=27sCv5YDQ1-fJFY5-2PF5w.1674763914814.7bc18717c5d120a55d9a46f114194f61&_x_zm_rhtaid=595) (0:42).

**Comment by TomChv** [comment]
So basically?

- **1**: Add logs to report what's happening: quick win
- **2**: Download faster (hard), do we have any part of the setup that can be enhanced to be faster? I feel that the slowest part is the downloading of the engine, which depends on the Network :/

**Comment by JammUtkarsh** [comment]
Hey @shykes
If this issue is not closed, I'm interested in contributing to the issue. 

I appreciate any information you can provide and look forward to contributing to the project.

Thank you,
Utkarsh Chourasia

**Comment by helderco** [comment]
> I believe that this is related to #4820. Once we have a PR that resolves that, we should also resolve this one. Both of these should be top of the **Ready for Work**.

According to the user feedback in the description, it's actually the same thing, not just related. It's the lack of feedback on auto-provisioning that's giving the impression that connect takes too long, so one of these should be a duplicate.

**Comment by shykes** [comment]
I think this has been mostly addressed?

- More clear feedback in the TUI, so you know something is happening
- Performance optimizations to make it incrementally faster than before.

wdyt @marcosnils @aluzzardi , ok to close?

**Comment by marcosnils** [comment]
> wdyt @marcosnils @aluzzardi , ok to close?

 yes, all performance aspects around session initialization / connect have been addressed AFAIK. SGTMt to close
